cpp/include/cudf/detail/utilities/rmm_host_vector.hpp

```diff
@@ -19,6 +19,7 @@
 #include <cudf/utilities/default_stream.hpp>
```

Comment:
I realize this is not a new file in this PR, but it's the first I've noticed it. I don't love the name `rmm_host_vector`, since this is not part of RMM, and RMM is an implementation detail. `cudf_host_vector` would be a better name. Or just `cudf::host_vector`.

Reply:
I'm not sure about this, but my impression was that the name was chosen because of the hope that the type will move to rmm at some point (CC @nvdbaranec ).
I'm fine with renaming to host_vector either way.

Reply:
Perhaps, but the existence of `rmm::device_vector` has already been questioned (since we discourage its use in RAPIDS). I suppose `host_vector` isn't as bad since its ctor doesn't need to launch a kernel and synchronize the device.

Reply:
We can certainly discuss adding a host_vector to RMM.

Reply:
I think we should rename to host_vector either way.
Also, `rmm_host_allocator` should probably be called something like `ref_host_allocator` since the use of the ref to type erase the resource type is the core of the type.

Reply:
renamed to `host_vector`.
kept `rmm_host_allocator` as such since it is "rmm-oriented" in the context of `thrust::host_vector` (only place where it's used).